File: app/src/main/java/com/hijamoya/example/dagger/injection/ControllerKey.java
package com.hijamoya.example.dagger.injection;

import com.bluelinelabs.conductor.Controller;

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

import dagger.MapKey;

@MapKey
@Target({ ElementType.METHOD }) public @interface ControllerKey {
    Class<? extends Controller> value();
}

File: app/src/main/kotlin/com/hijamoya/example/dagger/injection/ControllerKey.kt
package com.hijamoya.example.dagger.injection

import com.bluelinelabs.conductor.Controller

import dagger.MapKey
import kotlin.reflect.KClass
import kotlin.annotation.Target

@MapKey
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER,
        AnnotationTarget.PROPERTY_SETTER)
annotation class ControllerKey(val value: KClass<out Controller>)